dms_sample/stack.py

```diff
@@ -0,0 +1,292 @@
+    )
+
+
+def create_source_endpoint(stack: Stack, role: iam.Role, secret: secretsmanager.CfnSecret) -> dms.CfnEndpoint:
+    return dms.CfnEndpoint(
+            stack,
+            "source-postgres",
+            endpoint_type="source",
+            engine_name="aurora-postgresql",
+            database_name=DB_NAME,
+            postgre_sql_settings=dms.CfnEndpoint.PostgreSqlSettingsProperty(
+                secrets_manager_access_role_arn=role.role_arn,
+                secrets_manager_secret_id=secret.ref,
+            ),
+        )
+
+
+def create_secret(stack: Stack, host: str, db_port: int) -> secretsmanager.CfnSecret:
+    return secretsmanager.CfnSecret(
+            stack,
+            "postgres-secret",
+            secret_string=json.dumps(
+                {
+                    "username": USERNAME,
+                    "password": USER_PWD,
+                    "host": host,
+                    "port": db_port,
+                    "dbname": DB_NAME,
+                }
+            ),
+        )
+
+
+def create_security_group(stack: Stack, vpc: ec2.Vpc) -> ec2.SecurityGroup:
+    return ec2.SecurityGroup(
+        stack,
+        "sg",
+        vpc=vpc,
+        description="Security group for DMS sample",
+        allow_all_outbound=True,
+    )
+
+
+def create_kinesis_stream(stack: Stack, dms_assume_role: iam.Role) -> kinesis.Stream:
+    stream = kinesis.Stream(
+        stack, "TargetStream", shard_count=1, retention_period=cdk.Duration.hours(24)
+    )
+    stream.grant_read_write(dms_assume_role)
+    stream.apply_removal_policy(cdk.RemovalPolicy.DESTROY)
+    return stream
+
+
+def create_postgres_access_role(stack: Stack, postgres_secret: secretsmanager.CfnSecret) -> iam.Role:
+    return iam.Role(
+        stack,
+        "postgres-access-role",
+        assumed_by=iam.ServicePrincipal(f"dms.{stack.region}.amazonaws.com"),
+        inline_policies={
+            "AllowSecrets": iam.PolicyDocument(
+                statements=[
+                    iam.PolicyStatement(
+                        actions=["secretsmanager:GetSecretValue"],
+                        effect=iam.Effect.ALLOW,
+                        resources=[postgres_secret.ref],
+                    )
+                ]
+            )
+        },
+    )
+
+
+def create_kinesis_target_endpoint(stack: Stack, target_stream: kinesis.Stream, dms_assume_role: iam.Role) -> dms.CfnEndpoint:
+    return dms.CfnEndpoint(
+        stack,
+        "target",
+        endpoint_type="target",
+        engine_name="kinesis",
+        kinesis_settings=dms.CfnEndpoint.KinesisSettingsProperty(
+            stream_arn=target_stream.stream_arn,
+            message_format="json",
+            service_access_role_arn=dms_assume_role.role_arn,
+            include_control_details=True,
+            include_null_and_empty=True,
+            include_partition_value=True,
+            include_table_alter_operations=True,
+            include_transaction_details=True,
```

Comment:
we could probably set this one to `False` as we do not support it yet, I know we did it for other sources, but in this case I'd remove it 😄 as we're looking at all default values usually. 

Maybe we could add a note in the readme that this sample is using non default-values for the Kinesis target, and maybe make this parametrizable so that we could use default values instead? But this is a nice to have, I think it might make it easier for users that only have default values, might lead to less confusion. 

I think by default we might want to only use default values, and if a flag is set we create the target with the non-default values.